common/auto-updater/versions-db-manager.cpp

```diff
@@ -0,0 +1,307 @@
+            {   // Iterate through the version fields
+                std::unordered_map<std::string, std::string> curr_version;
+                for (auto it = ver.begin(); it != ver.end(); ++it)
+                {
+                    std::string element_key(it.key());
+
+                    auto schema_field(schema.find(element_key));
+                    if (schema_field != schema.end()) {
+                        if (it.value().is_string())
+                        {
+                            if (schema_field->second(it.value().get<std::string>()))
+                            {
+                                // Value validation passed - add to current version
+                                curr_version[element_key] = it.value().get<std::string>();
+                            }
+                            else
+                            {
+                                std::string error_str("Server versions file parsing error - " + element_key + " Validation fail on value : " + it.value().get<std::string>() + " \n");
+                            }
+                        }
+                        else
+                        {
+                            LOG_WARNING("Server versions file parsing error - " + element_key + " all values should be represented as a string \n");
+                            throw std::runtime_error("Server versions file parsing error - " + element_key + " - unknown field \n");
+                        }
+                    }
+                    else
+                    {
+                        LOG_WARNING("Server versions file parsing error - " + element_key + " - unknown field \n");
+                        throw std::runtime_error("Server versions file parsing error - " + element_key + "- unknown field \n");
+                    }
+                }
+
+                // Verify each key in the schema can be found in the current version fields
+                //std::pair<const K, E> &item
+                if (std::all_of(schema.cbegin(), schema.cend(), [curr_version](const std::pair<std::string, std::function<bool(const std::string&)>> &schema_item)
+                {
+                    return curr_version.end() != std::find_if(curr_version.cbegin(), curr_version.cend(), [schema_item](const std::pair<std::string, std::string> &ver_item) {return schema_item.first == ver_item.first; });
+                }))
+                {
+                    _server_versions_vec.emplace_back(curr_version); // Version added to valid versions vector
+                }
+                else
+                {
+                    LOG_WARNING("Server versions json file corrupted - not matching schema requirements \n");
+                    throw std::runtime_error("Server versions json file corrupted - not matching schema requirements \n");
+                }
+            }
+        }
+        else
+        {
+            LOG_WARNING("Server versions json file corrupted - Title not as expected\n");
```

Comment:
You can add what were the actual and the expected titles to make the corrective action

Reply:
Will be added

Reply:
Done